### PR TITLE
refactor(server): Use patch instead of update to resume/suspend

### PR DESCRIPTION
**Pull request by simster7**
Signed-off-by: Simon Behar <simbeh7@gmail.com>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/cron/controller.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func (cc *Controller) addCronWorkflowInformerHandler() {
 }
 
 func (cc *Controller) syncAll() {
-	log.Info("Syncing all CronWorkflows")
+	log.Debug("Syncing all CronWorkflows")
```
Comment (simster7) [inline workflow/cron/controller.go:208]: Change to `Debug` to avoid spamming the logs

File: server/cronworkflow/cron_workflow_server.go [hunk 2]
```diff
@@ -91,20 +93,19 @@ func (c *cronWorkflowServiceServer) DeleteCronWorkflow(ctx context.Context, req
...
+	return setCronWorkflowSuspend(true, req.Namespace, req.Name, ctx)
 }
 
-func (c *cronWorkflowServiceServer) setCronWorkflowSuspend(setTo bool, namespace, name string, ctx context.Context) (*v1alpha1.CronWorkflow, error) {
-	cwf, err := c.getCronWorkflowAndValidate(ctx, namespace, name, metav1.GetOptions{})
+func setCronWorkflowSuspend(setTo bool, namespace, name string, ctx context.Context) (*v1alpha1.CronWorkflow, error) {
+	data, err := json.Marshal(map[string]interface{}{"spec": map[string]interface{}{"suspend": setTo}})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to marshall cron workflow patch data")
```
Comment (alexec) [inline server/cronworkflow/cron_workflow_server.go:106]: return err? `fmt.Errorf("...:%w", err)`

Comment (simster7) [reply server/cronworkflow/cron_workflow_server.go:106]: Good catch

Comment (simster7) [reply server/cronworkflow/cron_workflow_server.go]: Done

File: workflow/cron/controller.go (hunk 1 above)
Comment (alexec) [reply workflow/cron/controller.go:208]: why do we log every 10s?

**Review by alexec — APPROVED** [review]

**Pull request merged**